test/unit/sdam/topology.test.js

```diff
@@ -530,9 +509,7 @@ describe('Topology (unit)', function () {
         //     returning their value
         //   - verify that 10 callbacks were called
 
-        topology.connect(err => {
-          expect(err).to.not.exist;
-
+        topology.connect().then(async () => {
```

Comment:
I don't think this is the correct conversion iiuc. Before we would call finish which calls done to complete the test. But this promise is floating, the test isn't blocked on it.